### PR TITLE
SCHED 337: Faults and EngTask files to OCSResourceService

### PR DESCRIPTION
**Pull request by stroncod**

### CODE REVIEW

File: scheduler/services/resource/night_resource_configuration.py [hunk 1]
```diff
@@ -2,14 +2,30 @@
...
+    end: datetime
+    time_loss = timedelta
+    comment: str
+
+@dataclass(frozen=True)
+class EngTask:
+    start: Optional[datetime]
+    end: Optional[datetime]
+    time_loss: Optional[timedelta]
+    comment: str
```
Comment (sraaphorst) [inline scheduler/services/resource/night_resource_configuration.py:26]: Since these have so much in common(ish), I was just wondering it there's any way to make them hierarchical with a common supertype? The `Optional`s though might not less us do it.

File: scheduler/services/resource/ocs_resource_service.py [hunk 2]
```diff
@@ -515,6 +532,86 @@ def none_to_str(value) -> str:
...
+                match = re.search(pattern, line)
+                if match:
+                    comment = match.group(1)
+                    rest_of_line = re.sub(pattern, '', line)
+                    _date, start_time, end_time = rest_of_line.strip().split()
+                    #  Single date for key
+                    just_date = datetime.strptime(_date, '%Y-%m-%d')
+                    # Time day in jd to calculate twilights
+                    time = Time(just_date)
+                    _, _, _, even_12twi, morn_12twi, _, _ = night_events(time,
```
Comment (sraaphorst) [inline scheduler/services/resource/ocs_resource_service.py:557]: This is kind of fragile in case we ever change the structure of `NightEvent` or the ordering of its elements.

Maybe we should do:
```python
ne = night_events(time, site.location, site.timezone)
eve_12twi = ne.twilight_evening_12
mor_12twi = ne.twilight_morning_12
```

File: scheduler/services/resource/ocs_resource_service.py [hunk 3]
```diff
@@ -515,6 +532,86 @@ def none_to_str(value) -> str:
...
+        with open(os.path.join(self._path, to_file), 'r') as file:
+            for line in file:
+                # Find pattern to keep bracket comments not splitted
+                match = re.search(pattern, line)
+                if match:
+                    comment = match.group(1)
+                    rest_of_line = re.sub(pattern, '', line)
+                    _date, start_time, end_time = rest_of_line.strip().split()
+                    #  Single date for key
+                    just_date = datetime.strptime(_date, '%Y-%m-%d')
```
Comment (sraaphorst) [inline scheduler/services/resource/ocs_resource_service.py:554]: Just curious why you're starting off local variable names with `_` here, especially if you're using them?

File: scheduler/services/resource/ocs_resource_service.py [hunk 4]
```diff
@@ -515,6 +532,86 @@ def none_to_str(value) -> str:
...
+                                       comment=comment)
+
+                    if just_date in self._eng_task[site]:
+                        self._eng_task[site][just_date].append(eng_task)
+                    else:
+                        self._eng_task[site][just_date] = [eng_task]
+                else:
+                    raise ValueError('Pattern not found. Format error on Eng Task file')
+
+    def parse_faults_file(self, site: Site, to_file: str)-> None:
```
Comment (sraaphorst) [inline scheduler/services/resource/ocs_resource_service.py:585]: Nitpick: Space missing in `)->`.

**Review by sraaphorst — COMMENTED** [review]
Looks good... just a couple small changes and questions. 👍

File: scheduler/services/resource/ocs_resource_service.py (hunk 2 above)
Comment (stroncod) [reply scheduler/services/resource/ocs_resource_service.py:557]: I agree. I change those.

File: scheduler/services/resource/night_resource_configuration.py (hunk 1 above)
Comment (stroncod) [reply scheduler/services/resource/night_resource_configuration.py:26]: I thought the same but in reality are not the same. The Optional type I think can be modified now that I'm getting the twilights at the moment of parsing.

**Review by sraaphorst — APPROVED** [review]
👍
